syntax = "proto3";
package draft.v1;

import "google/type/date.proto";
import "google/type/timeofday.proto";

service OrderService { 
    rpc CreateOrderFromDraft(CreateOrderRequest) 
        returns (OrderResponse);
}

message OrderItem {
    string product_id = 1;
    int64 quantity = 2;
    int64 subtotal_price = 3;
}

message CreateOrderRequest {
    string transaction_id = 1;

    string user_id = 2;
    string full_name = 3;
    string phone = 4;
    
    string location = 5;
    int64 total_sum = 6;

    google.type.Date delivery_date = 7;
    google.type.TimeOfDay slot_from = 8;
    google.type.TimeOfDay slot_to = 9;

    repeated OrderItem items = 10;
}

message OrderResponse {
    string order_id = 1;
    string status = 2;
}
